### PR TITLE
Adding Hubspot contacts sync app

### PR DESCRIPTION
**Pull request by gangeshwari-rajavelu**

### CODE REVIEW

File: Demo-Apps/hubspot_external_events_app/config/iparam_test_data.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+    "hubspot_url": "https://api.hubapi.com",
+    "hubspot_app_id" : "199193",
+    "hubspot_key": "8d68536e-2ec8-46be-9cd6-735f3481effe",
+    "hubspot_dev_key": "f4943b7f-4f2b-4b5b-b08e-5758e7f2c072",
+    "freshdesk_domain": "https://freshworkshelp576.freshdesk.com",
+    "freshdesk_key": "gm07e4GuWT4EGc7lZy6m"
```
Comment (hemchander23) [inline Demo-Apps/hubspot_external_events_app/config/iparam_test_data.json]: @gangeshwari-rajavelu I know this is dummy credentials. But can we rotate it and use fillers here like `<ENTER_YOUR_HUBSPOT_URL>` kinda :)

File: Demo-Apps/hubspot_external_events_app/config/iparams.json [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  "hubspot_app_id": {
+    "display_name": "Hubspot App Id",
+    "description": "Please enter your HubSpot App Id",
+    "type": "text",
+    "required": true
+  },
+  "hubspot_key": {
+    "display_name": "HubSpot API Key",
+    "description": "Please enter HubSpot API key (the API key associated with the regular hubspot account)",
+    "type": "text",
```
Comment (hemchander23) [inline Demo-Apps/hubspot_external_events_app/config/iparams.json]: Can we secure it ?

File: Demo-Apps/hubspot_external_events_app/config/oauth_config.json [hunk 3]
```diff
@@ -0,0 +1,11 @@
+
+{
+    "client_id": "af0ef631-472f-4ba6-96f8-c8e031ddd499",
+    "client_secret": "67203547-e05c-49a4-af66-0eb5a5e2e474",
```
Comment (hemchander23) [inline Demo-Apps/hubspot_external_events_app/config/oauth_config.json]: Let's replace this dummy creds with fillers.

File: Demo-Apps/hubspot_external_events_app/server/server.js [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+        isOAuth: true
+      };
+      $request.get(url, reqData)
+        .then(function (data) {
+          var contact = JSON.parse(data.response);
+          createFreshdeskContact(args.iparams.freshdesk_domain,
+            contact.properties.firstname.value,
+            contact.properties.email.value,
+            contact.properties.jobtitle.value);
+        }, function (err) {
```
Comment (hemchander23) [inline Demo-Apps/hubspot_external_events_app/server/server.js]: Let's stick to `then` / `catch` ? Just to make it more apparent

Comment (ravirajthedeveloper) [reply Demo-Apps/hubspot_external_events_app/server/server.js]: @hemchander23 I think the $request API works with 2 callback functions. The usage is explained the same way in the developer documentation [here](https://developer.freshdesk.com/v2/docs/request-api/#usage). We can go in this way for the Request API.

File: Demo-Apps/hubspot_external_events_app/config/iparam_test_data.json (hunk 1 above)
Comment (gangeshwari-rajavelu) [reply Demo-Apps/hubspot_external_events_app/config/iparam_test_data.json]: sure, i'll close this PR and raise a new one :)

**Pull request closed without merging**